README.md

```diff
@@ -56,50 +73,187 @@ and [Kevin Lacker](https://twitter.com/lacker)'s ([@Parse](http://parseplatform.
 + Appropriate to ***audience*** (*make it beginner friendly*...)
 + ***Opinionated*** (means people don't have to *think*)
 
+## But wait... what is an API, after all?
+
+The term `API` itself is quite generic and in layman's terms, 
+an API is just a **way of communicating with an interface programmatically**.
+It is a structured way for one program to offer services to other programs.
+
+Let's take the example of the 
+[`Twitter API`](https://developer.twitter.com/en/docs/platform-overview).
+We could use it to write a bot that displays the top tweets of the day.
+We, as users of the API, 
+don't need to know about the internal details
+of Twitter's systems, 
+nor does Twitter want us to.
+But using their API, we  are able to do specific things
+(like making a post or reading a timeline).
+**Twitter is exposing us features through an interface
+that we can consume**.
+
+If you've been around the web,
+you might have noticed that most of these APIs
+served by big companies like [`Google`](https://cloud.google.com/apis/docs/overview)
+or [`Microsoft`](https://developer.microsoft.com/en-us/graph)
+are often referred to as `REST API`s.
+
+Let's figure out what this is! 😉
+
 
 ## Types of API
 
+An API comes in different forms.
+When a developer get down to building an API,
+they first decide which specification to use.
+Most of the time, they go for `REST`.
+But, analysing product requirements, 
+a tech team may come to the conclusion 
+that calls for a different approach.
+
+`RESTful` APIs weren't always the *de facto* way of building APIs.
+In fact, there is an historical precedence to it.
+
+### RPC
+
+The earliest and simplest form of an API
+was **`RPC` Remote Procedure Call**,
+hailing from the 80's.
+
+The name speaks for itself.
+It's a straightforward interaction where a local client
+sends commands to a remote server.
+
+Here's a closer look at how it works.
+Both client and server use different call parameters,
+so they must be converted so it is understood by the other side.
+This conversion is made inside **stubs**.
+
+<img width="1418" alt="workflow RPC" src="https://user-images.githubusercontent.com/17494745/209990909-d2996bea-3162-4915-941d-944773505be4.png">
+
+`RPC` started with `XML` and later `JSON`-based versions.
+However, in 2015, Google created [`gRPC`](https://grpc.io/),
+a general-purpose `RPC` framework.
+Systems like Facebok's [`Apache Thrift`](https://thrift.apache.org/)
+or Twitch's [`Twirp`](https://blog.twitch.tv/en/2018/01/16/twirp-a-sweet-new-rpc-framework-for-go-5f2febbf35f/)
+use `gRPC` for internal communication between microservices.
+
+In these services, as *millions* of calls are made per-day,
```

Comment:
This is a _very_ long sentence. 

![image](https://user-images.githubusercontent.com/194400/212459867-2e946061-19e2-4661-8864-87b911750e25.png)

Please avoid writing sentences with more than 25 words as it dramatically lowers readability.
The average person would totally lose track of what they are reading half way into reading this.
Not saying that the people reading this doc are "stupid", they obvs aren't!
It's a highly technical doc aimed at an educated audience.
But just so you get into the habit of improving the readability of your writing for _all_ areas of your communication. Write shorter sentences. Thank me later. 👌